php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -1580,7 +1581,7 @@ public function serializeToJsonString($options = 0)
     /**
      * @ignore
      */
-    private function existField($field)
+    private function existField($field, $options = 0)
     {
         $getter = $field->getGetter();
         $hazzer = "has" . substr($getter, 3);
```

Comment:
Instead of checking for hazzers like this, this should actually use hasPresence method in FieldDescriptor.

Unfortunately, looks like `hasPresence` wasn't actually added to FieldDescriptor (should replace / deprecate the related but editions-hostile `hasOptionalKeyword`). We'll probably need to add / replace this: e.g. http://google3/third_party/java_src/protobuf/current/java/com/google/protobuf/Descriptors.java;l=1820;rcl=819963689 for java

Reply:
I'm slightly unclear, is this more of a project note for future work or a TODO for this PR? Felt a little like a project note so I wanted to double check.

Reply:
This is a TODO on our side. The current pattern hazzer pattern predates your change so this is probably fine for now -- we can clean this up on our side to use hasPresence (or hasOptionalKeyword in the interim) separately.